api/src/services/purging-config.js

```diff
@@ -0,0 +1,638 @@
+    const { pagination } = await act(getBatch, getIdsToPurge, roles, startKeyDocId, startKey);
+    startKey = pagination.nextKey;
+    startKeyDocId = pagination.nextKeyDocId;
+    nextBatch = pagination.nextBatch;
+  } while (nextBatch);
+};
+const countPurgedTasks = async (roles) => count(queryPurgedTasks, roles);
+
+const queryPurgedTargets = async (act, roles) => {
+  let startKeyDocId = 'target~';
+  let startKey = JSON.stringify(startKeyDocId);
+  let nextBatch;
+
+  const lastAllowedReportingIntervalTag = moment().subtract(TARGET_EXPIRATION_PERIOD, 'months').format('YYYY-MM');
+  // using `db.queryMedic` because PouchDB doesn't support `start_key_doc_id`
+  const getBatch = () => db.queryMedic('allDocs', {
+    limit: MAX_BATCH_SIZE,
+    start_key: JSON.stringify(startKeyDocId),
+    end_key: JSON.stringify(`target~${lastAllowedReportingIntervalTag}~`),
+  });
+
+  const getIdsToPurge = (rolesHashes, rows) => {
+    const toPurge = {};
+    rows.forEach(row => {
+      rolesHashes.forEach(hash => {
+        toPurge[hash] = toPurge[hash] || {};
+        toPurge[hash][row.id] = row.id;
+      });
+    });
+    return toPurge;
+  };
+
+  do {
+    logger.info(`Purging: Starting "targets" purge batch with id "${startKeyDocId}"`);
+    const { pagination } = await act(getBatch, getIdsToPurge, roles, startKeyDocId, startKey);
+    startKey = pagination.nextKey;
+    startKeyDocId = pagination.nextKeyDocId;
+    nextBatch = pagination.nextBatch;
+  } while (nextBatch);
+};
+const countPurgedTargets = async (roles) => count(queryPurgedTargets, roles);
+
+const getSchedule = config => {
+  if (!config) {
+    return;
+  }
+  if (config.text_expression) {
+    // text expression takes precedence over cron
+    return later.parse.text(config.text_expression);
+  }
+  if (config.cron) {
+    return later.parse.cron(config.cron);
+  }
+};
+
+const getNextRun = config => {
+  const schedule = getSchedule(config);
+  if (!schedule) {
+    return;
+  }
+  return later.schedule(schedule).next().toISOString();
+};
+
+const dryRun = async (appSettingsPurge) => {
+  const purgeFn = parsePurgeFn(appSettingsPurge.fn);
+  if (!purgeFn) {
+    throw new Error('Purging: No purge function configured.');
+  }
+
+  const roles = await getRoles();
+  await initPurgeDbs(roles);
+
+  const contacts = await countPurgedContacts(roles, purgeFn);
+  const unallocatedRecords = await countPurgedUnallocatedRecords(roles, purgeFn);
+  const tasks = await countPurgedTasks(roles);
+  const targets = await countPurgedTargets(roles);
+
+  const results = [contacts, unallocatedRecords, tasks, targets];
+  const wontChangeCount = sumBy(results, 'wontChangeCount');
+  const willPurgeCount = sumBy(results, 'willPurgeCount');
+  const willUnpurgeCount = sumBy(results, 'willUnpurgeCount');
+
+  const nextRun = getNextRun(appSettingsPurge);
+
+  return { wontChangeCount, willPurgeCount, willUnpurgeCount, nextRun };
+};
+
+module.exports = {
+  dryRun,
```

Comment:
curious about the expected performance for this endpoint? if purging takes several hours in production, how long would this endpoint take to execute? what implications will this have on replication performance?